py_wrap_generator.py

```diff
@@ -0,0 +1,2101 @@
+		self.name = name
+		self.namespace = None
+		self.link_type = link_type
+		self.id_ = id_
+		self.string_id = string_id
+		self.hash_id = hash_id
+		self.needs_clone = needs_clone
+		self.found_funs = []
+		self.found_vars = []
+		self.found_constrs = []
+
+	def printable_constrs(self):
+		ret = 0
+		for con in self.found_constrs:
+			if not con.protected:
+				ret += 1
+		return ret
+
+	def gen_decl(self, filename):
+		long_name = self.namespace + "::" + self.name
+
+		text = "\n\t// WRAPPED from " + filename
+		text += "\n\tstruct " + self.name
+		if self.link_type == link_types.derive:
+			text += " : public " + self.namespace + "::" + self.name
+		text += "\n\t{\n"
+
+		if self.link_type != link_types.derive:
+
+			text += "\t\t" + long_name + "* ref_obj;\n"
+
+			if self.link_type == link_types.ref_copy or self.link_type == link_types.pointer:
+				text += "\n\t\t" + long_name + "* get_cpp_obj() const\n\t\t{\n\t\t\treturn ref_obj;\n\t\t}\n"
+			elif self.link_type == link_types.global_list:
+				text += "\t\t" + self.id_.wtype.name + " " + self.id_.varname + ";\n"
+				text += "\n\t\t" + long_name + "* get_cpp_obj() const\n\t\t{"
+				text += "\n\t\t\t" + long_name + "* ret = " + long_name + "::get_all_" + self.name.lower() + "s()->at(this->" + self.id_.varname + ");"
+				text += "\n\t\t\tif(ret != NULL && ret == this->ref_obj)"
+				text += "\n\t\t\t\treturn ret;"
+				text += "\n\t\t\tthrow std::runtime_error(\"" + self.name + "'s c++ object does not exist anymore.\");"
+				text += "\n\t\t\treturn NULL;"
+				text += "\n\t\t}\n"
+
+			#if self.link_type != link_types.pointer:
+			text += "\n\t\tstatic " + self.name + "* get_py_obj(" + long_name + "* ref)\n\t\t{"
+			text += "\n\t\t\t" + self.name + "* ret = (" + self.name + "*)malloc(sizeof(" + self.name + "));"
+			if self.link_type == link_types.pointer:
+				text += "\n\t\t\tret->ref_obj = ref;"
+			if self.link_type == link_types.ref_copy:
+				if self.needs_clone:
+					text += "\n\t\t\tret->ref_obj = ref->clone();"
+				else:
+					text += "\n\t\t\tret->ref_obj = new "+long_name+"(*ref);"
+			if self.link_type == link_types.global_list:
+				text += "\n\t\t\tret->ref_obj = ref;"
+				text += "\n\t\t\tret->" + self.id_.varname + " = ret->ref_obj->" + self.id_.varname + ";"
+			text += "\n\t\t\treturn ret;"
+			text += "\n\t\t}\n"
+
+			if self.link_type == link_types.ref_copy:
+				text += "\n\t\tstatic " + self.name + "* get_py_obj(" + long_name + " ref)\n\t\t{"
+				text += "\n\t\t\t" + self.name + "* ret = (" + self.name + "*)malloc(sizeof(" + self.name + "));"
+				if self.needs_clone:
+					text += "\n\t\t\tret->ref_obj = ref.clone();"
+				else:
+					text += "\n\t\t\tret->ref_obj = new "+long_name+"(ref);"
+				text += "\n\t\t\treturn ret;"
+				text += "\n\t\t}\n"
+
+			if self.link_type != link_types.global_list:
+				text += "\n\t\t~" + self.name + "()\n\t\t{"
+				text += "\n\t\t\t//delete(this->ref_obj);\n\t\t}\n"
```

Comment:
This looks like it is generating commented out code?

Reply:
It did!
827a96d3a38afe025c9efbd182069a9c9adee267